cmd/status.go

```diff
@@ -0,0 +1,295 @@
+		if !found {
+			return errors.New("no package specified and package root not found")
+		}
+		if err != nil {
+			return errors.Wrap(err, "locating package root failed")
+		}
+		status, err = newLocalPackageStatus(packageRootPath, showAll)
+		if err != nil {
+			return err
+		}
+	}
+	if err := status.print(os.Stdout); err != nil {
+		return err
+	}
+	return nil
+}
+
+type packageStatus struct {
+	Name       string
+	Changelog  []packages.ChangeLogVersion
+	Local      *packages.PackageManifest
+	Production []packages.PackageManifest
+	Staging    []packages.PackageManifest
+	Snapshot   []packages.PackageManifest
+}
+
+func newLocalPackageStatus(packageRootPath string, showAll bool) (*packageStatus, error) {
+	manifest, err := packages.ReadPackageManifestFromPackageRoot(packageRootPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading package manifest failed")
+	}
+	changelog, err := packages.ReadChangelogFromPackageRoot(packageRootPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading package changelog failed")
+	}
+	status, err := newRemotePackageStatus(manifest.Name, showAll)
+	if err != nil {
+		return nil, err
+	}
+	status.Changelog = changelog
+	status.Local = manifest
+	return status, nil
+}
+
+func newRemotePackageStatus(packageName string, showAll bool) (*packageStatus, error) {
+	snapshotManifests, err := getDeployedPackage(packageName, snapshotURL, showAll)
+	if err != nil {
+		return nil, errors.Wrap(err, "retrieving snapshot deployment failed")
+	}
+	stagingManifests, err := getDeployedPackage(packageName, stagingURL, showAll)
+	if err != nil {
+		return nil, errors.Wrap(err, "retrieving staging deployment failed")
+	}
+	productionManifests, err := getDeployedPackage(packageName, productionURL, showAll)
+	if err != nil {
+		return nil, errors.Wrap(err, "retrieving production deployment failed")
+	}
+	return &packageStatus{
+		Name:       packageName,
+		Snapshot:   snapshotManifests,
+		Staging:    stagingManifests,
+		Production: productionManifests,
+	}, nil
+}
+
+func (p *packageStatus) pendingChanges() (*packages.ChangeLogVersion, error) {
+	if len(p.Changelog) == 0 || p.Local == nil {
+		return nil, nil
+	}
+	lastChangelogEntry := p.Changelog[0]
+	pendingVersion, err := semver.NewVersion(lastChangelogEntry.Version)
+	if err != nil {
+		return nil, err
+	}
+	currentVersion, err := semver.NewVersion(p.Local.Version)
+	if err != nil {
+		return nil, err
+	}
+	if currentVersion.LessThan(pendingVersion) {
+		return &lastChangelogEntry, nil
+	}
+	return nil, nil
+}
+
+func (p *packageStatus) print(w io.Writer) error {
+	changes, err := p.pendingChanges()
+	if err != nil {
+		return errors.Wrap(err, "parsing pending changelog entries failed")
+	}
+	bold := color.New(color.Bold)
+	red := color.New(color.FgRed, color.Bold)
+	cyan := color.New(color.FgCyan, color.Bold)
+	bold.Fprint(w, "Package: ")
+	cyan.Fprintln(w, p.Name)
+
+	environmentTable := [][]string{}
```

Comment:
```suggestion
	var environmentTable [][]string
```

Doesn't make much a difference, but I always prefer this format over the empty slice format. https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices There are a two more in the changes below.